icepack/utilities.py

```diff
@@ -46,21 +46,56 @@ def get_kwargs_alt(dictionary, keys, keys_alt):
                   for key, alt_key in zip(keys, keys_alt)))
 
 
-def facet_normal_2(mesh):
+def get_mesh_dimensions(mesh):
+    r"""Get the number of dimensions in the mesh. Half dimensions for extruded into the vertical.
+    """
+    dim = mesh.geometric_dimension()
+    if mesh.layers is not None:
+        dim -= 0.5
+    return dim
+
+def facet_normal_nd(mesh):
     r"""Compute the horizontal component of the unit outward normal vector
     to a mesh"""
-    ν = firedrake.FacetNormal(mesh)
-    return firedrake.as_vector((ν[0], ν[1]))
+    dim = get_mesh_dimensions(mesh)
+    if dim == 2:
+        return firedrake.FacetNormal(mesh)
+    elif dim == 2.5:
+        ν = firedrake.FacetNormal(mesh)
+        return firedrake.as_vector((ν[0], ν[1]))
+    elif dim in [1, 1.5]:
```

Comment:
There's no need for the final `elif` -- by that point it can only be the 1 or 1.5D model, so you can just unconditionally return `firedrake.FacetNormal(mesh)[0]`.